src/hash_table.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+from typing import Union
+import random
+
+
+class Node:
+    def __init__(self, key: str, val: Union[str, float], next: Node = None):
+        self.key = key
+        self.val = val
+        self.next = next
+
+
+class ChainHashTable:
+    def __init__(self, n: int):
+        self.__bucket = [Node(None, None)] * n
```

Comment:
Dont use this to generate list with type object. This will use the object `reference` for all elements of list. So when u change self.__bucket[0] then all the other self.__bucket[i] will be affected.
It's OK for [0] * size, but not [object type] * size